crates/bevy_core_pipeline/src/oit/mod.rs

```diff
@@ -0,0 +1,240 @@
+    OitResolvePlugin,
+};
+
+use crate::core_3d::{
+    graph::{Core3d, Node3d},
+    Camera3d,
+};
+
+/// Module that defines the necesasry systems to resolve the OIT buffer and render it to the screen
+pub mod resolve;
+
+/// Shader handle for the shader that draws the transparent meshes to the OIT layers buffer
+pub const OIT_DRAW_SHADER_HANDLE: Handle<Shader> = Handle::weak_from_u128(4042527984320512);
+
+/// Used to identify which camera will use OIT to render transparent meshes
+/// Alos used to configure OIT
+// TODO consider supporting multiple OIT techniques like WBOIT, Moment Based OIT,
+// depth peeling, stochastic transparency, ray tracing etc.
+// This should probably be done by adding an enum to this component
+#[derive(Component, Clone, Copy, ExtractComponent)]
+pub struct OrderIndependentTransparencySettings {
+    /// Controls how many layers will be used to compute the blending.
+    /// The more layers you use the more memory it will use but it will also give better results.
+    /// 8 is generally recommended, going above 16 is probably not worth it in the vast majority of cases
+    pub layer_count: u8,
+}
+
+impl Default for OrderIndependentTransparencySettings {
+    fn default() -> Self {
+        Self { layer_count: 8 }
+    }
+}
+
+/// Plugin needed to enable Order Independent Transparency
+pub struct OrderIndependentTransparencyPlugin;
+impl Plugin for OrderIndependentTransparencyPlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        load_internal_asset!(
+            app,
+            OIT_DRAW_SHADER_HANDLE,
+            "oit_draw.wgsl",
+            Shader::from_wgsl
+        );
+
+        app.add_plugins((
+            ExtractComponentPlugin::<OrderIndependentTransparencySettings>::default(),
+            OitResolvePlugin,
+        ))
+        .add_systems(Update, check_msaa)
+        .add_systems(Last, configure_depth_texture_usages);
+
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.add_systems(
+            Render,
+            prepare_oit_buffers.in_set(RenderSet::PrepareResources),
+        );
+
+        render_app
+            .add_render_graph_node::<ViewNodeRunner<OitResolveNode>>(Core3d, OitResolvePass)
+            .add_render_graph_edges(Core3d, (Node3d::MainTransparentPass, OitResolvePass));
+    }
+
+    fn finish(&self, app: &mut bevy_app::App) {
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.init_resource::<OitBuffers>();
+    }
+}
+
+// WARN This should only happen for cameras with the [`OrderIndependentTransparencySettings`]
+// but when multiple cameras are present on the same window
+// bevy reuses the same depth texture so we need to set this on all cameras.
+fn configure_depth_texture_usages(mut new_cameras: Query<&mut Camera3d, Added<Camera3d>>) {
+    for mut camera in &mut new_cameras {
```

Comment:
Because this is added as a default plugin to `CorePipelinePlugin`, it does need some kind of extra check to see if any cameras for this render target have added `OrderIndependentTransparencySettings`, otherwise this is effectively the same as changing the default to include texture binding.

Reply:
Right, it was fine for us at foresight, but it's less fine for this 😅. I'll add a check for the render target.